src/MarketplaceAppProvider.tsx

```diff
@@ -25,18 +25,16 @@ export const MarketplaceAppProvider: React.FC<ProviderProps> = ({ children }) =>
 
   // Initialize the SDK and track analytics event
   useEffect(() => {
-    (async () => {
-      try {
-        const appSdk: Extension = await ContentstackAppSDK.init();
+    try {
+      ContentstackAppSDK.init().then(async (appSdk: Extension) => {
         setAppSdk(appSdk);
-
         const appConfig: KeyValueObj = await appSdk.getConfig();
         setConfig(appConfig);
-      } catch (err) {
-        setFailed(true);
-      }
-    })();
-  }, [setFailed, setAppSdk, setConfig]);
+      });
```

Comment:
add a catch here for promise. It will never go into catch block. ?You can get rid of try catch